app/src/main/java/fr/free/nrw/commons/explore/depictions/DepictsClient.java

```diff
@@ -0,0 +1,186 @@
+        title = title.replace(" ", "_");
+        String MD5Hash =  getMd5(title);
+        /**
+         * We use 70 pixels as the size of our Thumbnail (as it is the perfect fits our UI)
+         */
+        return baseUrl + MD5Hash.charAt(0) + '/' + MD5Hash.charAt(0) + MD5Hash.charAt(1) + '/' + title + "/70px-" + title;
+    }
+
+    /**
+     * Ex: entityId = Q357458
+     * value returned = Elgin Baylor Night program.jpeg
+     */
+    public Single<String> getP18ForItem(String entityId) {
+        return depictsInterface.getImageForEntity(entityId)
+                .map(commonsFilename -> {
+                    String name;
+                    try {
+                        JsonObject claims = commonsFilename.getAsJsonObject("claims").getAsJsonObject();
+                        JsonObject p18 = claims.get("P18").getAsJsonArray().get(0).getAsJsonObject();
+                        JsonObject mainsnak = p18.get("mainsnak").getAsJsonObject();
+                        JsonObject datavalue = mainsnak.get("datavalue").getAsJsonObject();
+                        JsonPrimitive value = datavalue.get("value").getAsJsonPrimitive();
+                        name = value.toString();
+                        name = name.substring(1, name.length() - 1);
+                    } catch (Exception e) {
+                        name="";
+                    }
+                    if (!name.isEmpty()){
+                        return getThumbnailUrl(name);
+                    } else return NO_DEPICTED_IMAGE;
+                })
+                .singleOrError();
+    }
+
+    /**
+     * @return list of images for a particular depict entity
+     */
+    public Observable<List<Media>> fetchImagesForDepictedItem(String query, int limit, int sroffset) {
+        return mediaInterface.fetchImagesForDepictedItem("haswbstatement:" + BuildConfig.DEPICTS_PROPERTY + "=" + query, String.valueOf(sroffset))
+                .map(mwQueryResponse -> {
+                    List<Media> mediaList =  new ArrayList<>();
+                    for (Search s: mwQueryResponse.getQuery().getSearch()) {
+                        Media media = new Media(null,
+                                getUrl(s.getTitle()),
+                                s.getTitle(),
+                                new HashMap<>(),
+                                "",
+                                0,
+                                safeParseDate(s.getTimestamp()),
+                                safeParseDate(s.getTimestamp()),
+                                ""
+                        );
+                        mediaList.add(media);
+                    }
+                    return mediaList;
+                });
+    }
+
+    /**
+     * Get url for the image from media of depictions
+     * Ex: Tiger_Woods
+     * Value: https://upload.wikimedia.org/wikipedia/commons/thumb/6/67/Tiger_Woods.jpg/70px-Tiger_Woods.jpg
+     */
+    private String getUrl(String title) {
```

Comment:
IMO this is not the right way of getting the image url. It could lead to errors. I remember running into issues with such URL formations in media details screen.